neuralmonkey/trainers/regularizers.py

```diff
@@ -104,36 +118,35 @@ def __init__(self,
                 on the previous task.
         """
         check_argument_types()
-
         Regularizer.__init__(self, name, weight)
 
-        if gradients_file is None:
-            raise ValueError("Missing gradients_file")
-        if variables_file is None:
-            raise ValueError("Missing variables_file")
-
-        log("Loading initial variables for EWC from {}".format(variables_file))
+        log("Loading initial variables for EWC from "
+            "{}.".format(variables_file))
         self.init_vars = tf.contrib.framework.load_checkpoint(variables_file)
-        log("EWC initial variables loaded")
+        log("EWC initial variables loaded.")
 
-        log("Loading gradient estimates from {}".format(gradients_file))
+        log("Loading gradient estimates from {}.".format(gradients_file))
         self.gradients = np.load(gradients_file)
-        log("Gradient estimates loaded")
+        log("Gradient estimates loaded.")
 
-    def value(self, variables: List[tf.Tensor]) -> float:
+    def value(self, variables: List[tf.Tensor]) -> tf.Tensor:
         ewc_value = tf.constant(0.0)
         for var in variables:
-            var_name = var.name.split(":")[0]
+            var_name = var.name
+            init_var_name = var_name.split(":")[0]
             if (var_name in self.gradients.files
-                    and self.init_vars.has_tensor(var_name)):
-                init_var = self.init_vars.get_tensor(var_name)
-                gradient = tf.constant(
-                    self.gradients[var_name], name="ewc_gradients")
+                    and self.init_vars.has_tensor(init_var_name)):
+                init_var = tf.constant(
+                    self.init_vars.get_tensor(init_var_name),
+                    name="{}_init_value".format(init_var_name))
+                grad_squared = tf.constant(
+                    np.square(self.gradients[var_name]),
+                    name="{}_ewc_weight".format(init_var_name))
```

Comment:
Teď si ale právě nepojmenováváš v grafu gradienty, ale jejich druhý mocniny. To jsi mohl docílit už tím, že bys napsal `tf.square(gradient, name="kráva")`

Reply:
nevim, takhle ale uz do grafu pridavam predpociane druhe mocniny... v opacnem pripade (s tf.square) by se ten vypocet provadel pokazde, kdyz bys potreboval ten vysledek

ve vysledku tam asi takovy rozdil v rychlosti nebude, ale prijde mi to rozumnejsi si konstanty predpocitat, nez je fouknes do grafu